apps/omg/lib/omg/input_pointer.ex

```diff
@@ -14,33 +14,132 @@
+
+  # TODO(achiurizo)
+  # Do we really need this? How does this work with output type marker?
+  @input_pointer_type_marker <<1>>
+
+  @type utxo_pos_tuple() :: {
+          :utxo_position,
+          # blknum
+          non_neg_integer,
+          # txindex
+          non_neg_integer,
+          # oindex
+          non_neg_integer
+        }
+
+  @type utxo_pos_db_tuple() :: {
+          # blknum
+          non_neg_integer,
+          # txindex
+          non_neg_integer,
+          # oindex
+          non_neg_integer
+        }
+
+  @type db_key_tuple() :: {
+          :input_pointer,
+          <<_::8>>,
+          utxo_pos_db_tuple()
+        }
+
+  @position_too_low_error_tuple {:error, :encoded_utxo_position_too_low}
+  @type position_too_low_error_tuple() :: {:error, :encoded_utxo_position_too_low}
 
-  This module specificially dispatches generic calls to the various specific types
+  @doc """
+  Decode an integer into a utxo position tuple.
+
+  ## Examples
+
+    iex> OMG.InputPointer.decode!(1000020003)
+    {:utxo_position, 1, 2, 3}
   """
-  @input_pointer_types_modules Application.fetch_env!(:omg, :input_pointer_types_modules)
+  @spec decode!(integer() | <<_::256>>) :: utxo_pos_tuple()
+  def decode!(encoded) do
+    {:ok, decoded} = decode(encoded)
+    decoded
+  end
 
-  def from_db_key({:input_pointer, type_marker, db_value}),
-    do: @input_pointer_types_modules[type_marker].from_db_key(db_value)
+  # TODO(achiurizo)
+  # Do we really need an error tuple for a value too low?
+  @doc """
+  Decode an integer into a utxo position tuple.
 
-  # default clause for backwards compatibility
-  def from_db_key(db_value), do: OMG.InputPointer.UtxoPosition.from_db_key(db_value)
-end
+  ## Examples
 
-defprotocol OMG.InputPointer.Protocol do
-  @moduledoc """
-  Captures the varying behavior of how the outputs can be pointed to in child chain transactions
+    # Returns an utxo position tuple from an integer
+    iex> OMG.InputPointer.decode(1000020003)
+    {:ok, {:utxo_position, 1, 2, 3}}
+
+    # Returns an utxo position tuple from a RLP-encodable binary.
+    iex> OMG.InputPointer.decode(<<0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 119, 53, 187, 17>>)
+    {:ok, {:utxo_position, 2, 1, 1}}
+
+    # Returns an error tuple if the value is too low
+    iex> OMG.InputPointer.decode(-1)
+    {:error, :encoded_utxo_position_too_low}
   """
+  @spec decode(integer() | <<_::256>>) :: {:ok, utxo_pos_tuple()} | position_too_low_error_tuple()
+  def decode(encoded) when is_binary(encoded) and byte_size(encoded) == 32,
+    do: encoded |> :binary.decode_unsigned(:big) |> decode()
+
+  def decode(encoded) when is_integer(encoded) and encoded <= 0,
+    do: @position_too_low_error_tuple
+
+  def decode(encoded) when is_integer(encoded) do
+    %ExPlasma.Utxo{blknum: blknum, txindex: txindex, oindex: oindex} = ExPlasma.Utxo.new(encoded)
+    {:ok, {:utxo_position, blknum, txindex, oindex}}
+  end
```

Comment:
Eventually, I'd like to remove this call as well, but because I'm unsure where we care about this error tuple, I'm leaving it in another follow up PR(or if comments suggest otherwise) to clean this portion up.

Reply:
oh I think you changed the code and the comment is dangling without proper context now